### PR TITLE
feat: [PL-28361]: Adding All tab in Select Secret modal

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

Screenshot 2022-10-11 at 12 47 55 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 18 Issues 
* Code issue 18 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11663)

File: src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx [hunk 1]
```diff
@@ -461,41 +466,30 @@ export function getReferenceFieldProps({
...
-              ...(!category && { type }),
-              ...gitFilterParams,
-              category,
-              searchTerm: search,
-              pageIndex: page,
-              pageSize: 10,
-              projectIdentifier: scope === Scope.PROJECT ? projectIdentifier : undefined,
-              orgIdentifier: scope === Scope.PROJECT || scope === Scope.ORG ? orgIdentifier : undefined
-            }
-          })
```
Comment (meenakshi-harness) [inline src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx:498]: Can we revert this change in this PR. It will need extra testing for Connector reference.

File: src/modules/40-gitsync/components/gitSyncRepoForm/__tests__/GitSyncRepoForm.test.tsx [hunk 2]
```diff
@@ -80,7 +80,7 @@ jest.mock('services/cd-ng', () => {
     }),
     usePostGitSync: jest.fn().mockImplementation(() => ({ mutate: createGitSynRepo })),
     useGetConnector: jest.fn().mockImplementation(() => ({ data: gitHubMock, refetch: getGitConnector })),
-    getConnectorListPromise: jest.fn().mockImplementation(() => Promise.resolve(gitHubMock)),
```
Comment (meenakshi-harness) [inline src/modules/40-gitsync/components/gitSyncRepoForm/__tests__/GitSyncRepoForm.test.tsx:83]: same .. We can only make changes that are needed for Secret Reference component in this PR?

File: src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx:498]: I have changed enum in the common component, hence changes are required here as well

Comment (meenakshi-harness) [reply src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx:498]: Yes I meant the changed for removing v1 for connector list api. I see those are reverted now. lgtm.

**Review by meenakshi-harness — DISMISSED** [review]
